qiskit/algorithms/optimizers/de.py

```diff
@@ -0,0 +1,134 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Differential Evolution (DE) optimizer"""
+
+from typing import Callable, Tuple, List, Dict, Union
+
+import numpy as np
+from scipy.optimize import differential_evolution
+from .optimizer import Optimizer, OptimizerSupportLevel
+
+
+class DE(Optimizer):
+    """
+    Differential Evolution (DE) optimizer
+
+    DE is a global and derivative-free optimizer that seeks
+    to find the optimal value of an objective function through stochastically
+    searching the optimization space of possible solutions.
+    DE belongs to the family of evolutionary optimization techniques
+    such as genetic and particle swarm algorithms. It is best suited
+    for combinatorial optimization problems and operates on real-parameter
+    and real-valued objective functions.
+
+    Specifically, the algorithm begins with an initial population
+    of candidate solutions (can be random or user-specified) and new candidates
+    are iteratively created through a combination of mutation, recombination
+    and stochastic selection. The main parameters that govern the optimization
+    include the evolutionary strategy, population size, mutation constant
+    (differential weight) and recombination factor (crossover probability).
+    The algorithm terminates if either the maximum number of iterations/generations
+    have been reached or the standard deviation of the generation of candidates
+    is below a user-specified threshold.
+
+    Uses scipy.optimize.differential_evolution
+    For further detail, please refer to
+    https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.differential_evolution.html
+    """
+
+    _OPTIONS = ['strategy', 'maxiter', 'popsize', 'mutation',
+                'recombination', 'init', 'tol', 'atol',
+                'polish', 'disp', 'workers']
+
+    def __init__(self,
+                 strategy: str = 'currenttobest1bin',
+                 maxiter: int = 10,

```

Comment:
Could you change `maxiter` to be the first argument? It's a minor thing but would be more similar to the other optimizers 🙂

Reply:
Thanks for picking up on this. Will do!

Reply:
Fixed!